src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -112,14 +113,17 @@ void TaskRevolutionParameters::fillAxisCombo(bool forceRefill)
     if (forceRefill){
         ui->axis->clear();
 
-        for(size_t i = 0; i < axesInList.size(); i++){
+        int i = 0;

```

Comment:
That's a weird way iterating over the array.

I recommend to replace `std::vector<App::PropertyLinkSub*> axesInList;` with
`std::vector<std::unique_ptr<App::PropertyLinkSub>> axesInList;` and adjust the implementation accordingly. Then you don't have to iterate over the array when clearing it.

Reply:
Done, but this is a new C++ field for me (never used std::make_unique yet), can you therefore please check if I did it right?